packages/teams-js/src/private/files.ts

```diff
@@ -710,40 +731,90 @@ export namespace files {
-   * Download Action : Allows downloading file(s) from the given 3P cloud storage provider
-   * Delete Action : Allows deleting file(s) from the given 3P cloud storage provider
+   * Initiates the download 3P cloud storage file(s) flow, which will download existing file(s) from the given 3P provider
    *
-   * @param cloudStorageProviderFileActionRequest 3P cloud storage provider file action (Upload / Download / Delete) request content
-   * @param callback Callback that will be triggered post 3P cloud storage action
+   * @param downloadFileRequest 3P cloud storage provider download file(s) action request content
+   * @param callback Callback that will be triggered post downloading existing file(s) flow is finished
    */
-  export function performCloudStorageProviderFileAction(
-    cloudStorageProviderFileActionRequest: CloudStorageProviderRequest<CloudStorageProviderActionContent>,
+  export function downloadCloudStorageProviderFile(
+    downloadFileRequest: CloudStorageProviderRequest<CloudStorageProviderDownloadFileContent>,
     callback: (error?: SdkError) => void,
   ): void {
     ensureInitialized(FrameContexts.content);
 
     if (!callback) {
       throw getSdkError(
         ErrorCode.INVALID_ARGUMENTS,
-        '[files.performCloudStorageProviderFileAction] callback cannot be null',
+        '[files.downloadCloudStorageProviderFile] callback cannot be null',
       );
     }
 
-    if (!(cloudStorageProviderFileActionRequest && cloudStorageProviderFileActionRequest.content)) {
+    if (!(downloadFileRequest && downloadFileRequest.content)) {
       throw getSdkError(
         ErrorCode.INVALID_ARGUMENTS,
-        '[files.performCloudStorageProviderFileAction] 3P cloud storage provider request content is missing',
+        '[files.downloadCloudStorageProviderFile] 3P cloud storage provider request content is missing',
       );
     }
 
-    sendMessageToParent(
-      'files.performCloudStorageProviderFileAction',
-      [cloudStorageProviderFileActionRequest],
-      callback,
-    );
+    sendMessageToParent('files.downloadCloudStorageProviderFile', [downloadFileRequest], callback);
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Initiates the upload 3P cloud storage file(s) flow, which will upload file(s) to the given 3P provider
+   *
+   * @param uploadFileRequest 3P cloud storage provider upload file(s) action request content
+   * @param callback Callback that will be triggered post uploading file(s) flow is finished
+   */
+  export function uploadCloudStorageProviderFile(
+    uploadFileRequest: CloudStorageProviderRequest<CloudStorageProviderUploadFileContent>,
+    callback: (error?: SdkError) => void,
+  ): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!callback) {
+      throw getSdkError(ErrorCode.INVALID_ARGUMENTS, '[files.uploadCloudStorageProviderFile] callback cannot be null');
+    }
+
+    if (!(uploadFileRequest && uploadFileRequest.content)) {
+      throw getSdkError(
```

Comment:
validate itemList

Reply:
(also need to validate destination folder: make sure it's a directory, has valid params on it, etc.

Reply:
Done. Made validation and restriction on itemList and destination folder.